File: server/server-core/src/main/java/com/simple/castle/server/physic/unit/GhostObject.java
package com.simple.castle.server.physic.unit;

import com.badlogic.gdx.physics.bullet.collision.btCollisionObject;
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody;

public class GhostObject extends PhysicObject {
    public GhostObject(final btRigidBody.btRigidBodyConstructionInfo constructionInfo) {
        super(constructionInfo);
        this.body.setCollisionFlags(this.body.getCollisionFlags() | btCollisionObject.CollisionFlags.CF_NO_CONTACT_RESPONSE);
    }
}

File: server/server-core/src/main/java/castle/server/ashley/physic/GhostObject.kt
package castle.server.ashley.physic

import com.badlogic.gdx.physics.bullet.collision.btCollisionObject
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody

open class GhostObject(constructionInfo: btRigidBody.btRigidBodyConstructionInfo) : PhysicObject(constructionInfo) {
    init {
        body.collisionFlags = body.collisionFlags or btCollisionObject.CollisionFlags.CF_NO_CONTACT_RESPONSE
    }
}